artemis-commons/src/main/java/org/apache/activemq/artemis/utils/CompositeAddress.java

```diff
@@ -41,12 +41,36 @@ public static SimpleString toFullyQualified(SimpleString address, SimpleString q
       return result;
    }
 
+   /**
+    * Checks whether a given string complies the fully qualified queue name (FQQN) standard or not.
+    *
+    * @param address The string to check for FQQN standard compliance.
+    * @return boolean
+    */
    public static boolean isFullyQualified(String address) {
-      return address == null ? false : address.contains(SEPARATOR);
+      if (address == null) {
+         return false;
+      }
+
+      // If address length is incapable of holding the separator and two more characters
+      if (address.length() < SEPARATOR.length() + 2) {
```

Comment:
Sicne the separator is a constant, the right hand side here is effectively a constant, "2+2=4" in this case. @franz1981 Will the JVM quickly figure that out since the SEPARATOR constant value is a String and thus final, or would declaring a constant for the calculation make sense?

Reply:
I implemented this, creating a new class constant which I even reuse in the return statement thus making the function idempotent to changes to the SEPARATOR I believe! ;)